debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorMetadata.java

```diff
@@ -0,0 +1,153 @@
+        additionalMetadata.put(MongoDbConnectorConfig.SIGNAL_DATA_COLLECTION.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.QUERY_FETCH_SIZE.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.MAX_BATCH_SIZE.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.MAX_QUEUE_SIZE.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.MAX_QUEUE_SIZE_IN_BYTES.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.POLL_INTERVAL_MS.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.RETRIABLE_RESTART_WAIT.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+        additionalMetadata.put(MongoDbConnectorConfig.SOURCE_STRUCT_MAKER_VERSION.name(), new PropertyMetadata(false, ConnectorProperty.Category.ADVANCED));
+
+        MONGODB_PROPERTIES = Collections.unmodifiableMap(additionalMetadata);
+    }
+
+    protected ConnectionContext.MongoPrimary primary(MongoDbTaskContext context) throws Throwable {
+        ReplicaSet replicaSet = ReplicaSet.parse(context.getConnectionContext().hosts());
+        return context.getConnectionContext().primaryFor(replicaSet, context.filters(), (s, throwable) -> {
+            throw new DebeziumException(s, throwable);
+        });
+    }
+
+    protected <T extends DataCollectionId> Stream<T> determineDataCollectionsToBeSnapshotted(
+                                                                                             CommonConnectorConfig connectorConfig,
+                                                                                             final Collection<T> allDataCollections) {
+        final Set<String> snapshotAllowedDataCollections = connectorConfig.getDataCollectionsToBeSnapshotted();
+        if (snapshotAllowedDataCollections.size() == 0) {
+            return allDataCollections.stream();
+        }
+        else {
+            return allDataCollections.stream()
+                    .filter(dataCollectionId -> snapshotAllowedDataCollections.stream().anyMatch(s -> dataCollectionId.identifier().matches(s)));
+        }
+    }
+
+    @Override
+    public ConnectorDescriptor getConnectorDescriptor() {
+        return new ConnectorDescriptor("mongodb", "Mongo DB", getConnector().version());
+    }
+
+    @Override
+    public Map<String, PropertyMetadata> allPropertiesWithAdditionalMetadata() {
+        return MONGODB_PROPERTIES;
+    }
+
+    // @TODO using this vs. #getConnector().config().configKeys()
```

Comment:
Either we could remove `#getAllConnectorFields()` and use `#getConnector().config() / .configKeys()` instead or always use `#getAllConnectorFields()`?

Currently `#getConnector().config()...` is works with ConfigDef (dependency to Kafka's connect-api package), `#getAllConnectorFields()` return Field.Set from Debezium Core.

Reply:
I now tend to use the `#getAllConnectorFields()` to avoid the dependency to Kafka Connect classes.